src/languageservice/services/yamlSchemaService.ts

```diff
@@ -82,6 +83,53 @@ export class FilePatternAssociation {
   }
 }
 
+type MatchFn = (str: string) => boolean;
+export class JSONFilePatternAssociation {
+  private readonly matchers: MatchFn[];
+  private readonly isInclude: boolean[];
+  private readonly uris: string[];
+
+  constructor(pattern: string[], uris: string[]) {
+    this.isInclude = [];
+    this.matchers = [];
+    try {
+      for (let p of pattern) {
+        const include = p[0] !== '!';
+        if (!include) {
+          p = p.substring(1);
+        }
+
+        if (!p.startsWith('**/')) {
```

Comment:
I've been playing around with it a bit and it seems to work pretty well. I did find a few issues, mainly https://github.com/ansible-community/vscode-ansible/issues/36#issuecomment-790485417 wasn't resolving correctly. I tried with this patch and everything seemed to work okay after that:

```suggestion
        if (!p.startsWith('**/')) {
          if (p.startsWith('/')) {
            p = p.substring(1);
          }
```